src/hooks/useReactorToken.ts

```diff
@@ -0,0 +1,92 @@
+import { useUserStore } from 'context/User/useUserContextStore';
+import { decodeJwt, JWTPayload } from 'jose';
+import useSWR from 'swr';
+import { authorizeTask, formatEndpointAddress } from 'utils/dataPlane-utils';
+import { getURL, hasLength } from 'utils/misc-utils';
+
+interface Token {
+    url: URL | null;
+    parsed: JWTPayload;
+    token: string;
+}
+
+type FetcherArgs = [string, string, string | undefined];
+const fetcher = async ({
+    0: catalogName,
+    1: sessionKey,
+}: FetcherArgs): Promise<Token> => {
+    const response = await authorizeTask(sessionKey, catalogName);
+
+    if (
+        !hasLength(response.reactorToken) ||
+        !hasLength(response.reactorAddress)
+    ) {
+        throw new Error('data plane auth token response was empty');
+    }
+    const { reactorAddress, reactorToken } = response;
+
+    const parsed = decodeJwt(reactorToken);
+
+    return {
+        token: reactorToken,
+        url: getURL(formatEndpointAddress(reactorAddress)),
+        parsed,
+    };
+};
+
+// Returns the refresh interval in milliseconds for a given auth token response. SWR will re-fetch
+// this token automatically after this refresh interval, so we determine it based on the expiration
+// of the JWT.
+const getTokenRefreshInterval = (token: Token | undefined): number => {
+    if (!token) {
+        return 0;
+    }
+
+    try {
+        if (!token.parsed.exp) {
+            throw new Error('data plane JWT is missing exp');
+        }
+        // JWT date is in seconds, while Date.now and SWR use millis.
+        const exp = token.parsed.exp * 1000;
+        const now = Date.now();
+        if (now >= exp) {
+            console.error('data plane JWT is expired', {
+                now,
+                token,
+            });
```

Comment:
We should move this over to a LR console log

Reply:
I _knew_ there was something I had missed. Will update.